src/tests/Search.test.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+import { Router } from 'react-router-dom';
+import { render, fireEvent } from '@testing-library/react';
+import { createMemoryHistory } from 'history';
+import '@testing-library/jest-dom/extend-expect';
+import App from '../App';
+import defaultSubreddit from '../helpers/constants';
+
+describe('Search', () => {
+  it('input to be defaultSubreddit when clicking on Search link', () => {
+    const history = createMemoryHistory();
+    history.push('/search/wholesomememes');
+
+    const { getByRole } = render(
+      <Router history={history}>
+        <App />
+      </Router>,
+    );
+
+    const searchInput = getByRole('textbox');
+    const searchLink = getByRole('link', { name: /Search/ });
+    fireEvent.click(searchLink);
+
+    expect(searchInput.value).toBe(defaultSubreddit);
+  });
+
+  it('input has subreddit from URL', () => {
+    const history = createMemoryHistory();
+    history.push('/search/CasualKnitting');
+
+    const { getByRole } = render(
+      <Router history={history}>
+        <App />
+      </Router>,
+    );
+
+    const searchInput = getByRole('textbox');
+
+    expect(searchInput.value).toBe('CasualKnitting');
+  });
+
+  it('input changes when new subreddit is entered', () => {
+    const history = createMemoryHistory();
+    history.push('/search');
+
+    const { getByRole } = render(
+      <Router history={history}>
+        <App />
+      </Router>,
+    );
```

Comment:
This code seems to repeat. Kent C Dodds has a nice blog post [AHA testing](https://kentcdodds.com/blog/aha-testing). I guess that can be applied here. You can extract this code into a `setup` or `renderSearchPage` function that returns the queries returned by `render`.

If you like you can also return additional values and functions like `searchInput` or `changeSearchInput` as explained in one of the [last sections](https://kentcdodds.com/blog/aha-testing#aha-testing-with-react)

Reply:
I thought I had tried to abstract it out in an earlier set of tests and it didn't work, but it seems to be fine now.  Though I don't think I can use it if I need to access the history prop like in the last test?